tasks/main.yml

```diff
@@ -6,43 +6,34 @@
-  register: cluster_template_show
-  changed_when: False
-  failed_when: False
+- name: Set a fact about the Ansible python interpreter
+  set_fact:
+    old_ansible_python_interpreter: "{{ ansible_python_interpreter | default('/usr/bin/python' + ansible_python.version.major | string) }}"
 
 - name: Ensure container cluster templates exist
-  command: >
-    {{ os_container_clusters_venv ~ '/bin/' if os_container_clusters_venv else '' }}openstack
-    {% for auth_name, auth_value in os_container_clusters_auth.items() %}
-    --os-{{ auth_name | replace('_', '-') }}='{{ auth_value }}'
-    {% endfor %}
-    {% if os_container_clusters_cacert is defined %}
-    --os-cacert {{ os_container_clusters_cacert }}
-    {% endif %}
-    --os-interface=public
-    coe cluster template create {{ item.item.name }}
-    {% for name, value in item.item.items() %}
-    {% if name not in ['name', 'labels'] %}
-    --{{ name | replace('_', '-') }}{% if value is not none %}='{{ value }}'{% endif %}
-    {% elif name == 'labels' %}
-    {% if value is mapping %}
-    --labels='{{ item.item['labels'].items() | map('join', '=') | join(',') }}'
-    {% else %}
-    --labels='{{ value }}'
-    {% endif %}
-    {% endif %}
-    {% endfor %}
-  with_items: "{{ cluster_template_show.results }}"
-  when: item.rc != 0
+  vars:
+    ansible_python_interpreter: "{{ os_container_clusters_venv ~ '/bin/python' if os_container_clusters_venv != None else old_ansible_python_interpreter }}"
+  openstack.cloud.coe_cluster_template:
+    auth: "{{ os_container_clusters_auth }}"
+    auth_type: "{{ os_container_clusters_auth_type }}"
+    cacert: "{{ os_container_clusters_cacert | default(omit) }}"
+    interface: "{{ os_container_clusters_interface | default(omit, true) }}"
+    labels: "{{ item.labels | default(omit) }}"
+    external_network_id: "{{ item.external_network_id | default(undef(hint = 'You must provide an external network'))}}"
+    master_flavor_id: "{{ item.master_flavor | default(undef(hint = 'You must provide a master flavor'))}}"
+    flavor_id: "{{ item.flavor | default(undef(hint = 'You must provide a flavor')) }}"
+    image_id: "{{ item.image | default(undef(hint = 'You must provide an image')) }}"
+    name: "{{ item.name | default(undef(hint = 'You must provide a name')) }}"
+    coe: "{{ item.coe | default(undef(hint = 'You must provide a coe')) }}"
+    network_driver: "{{ item.network_driver | default(omit) }}"
+    master_lb_enabled: "{{ item.master_lb_enabled |default(True) | bool }}"
+    floating_ip_enabled: "{{ item.floating_ip_enabled | default(True) | bool }}"
+    dns_nameserver: "{{ item.dns_nameserver | default(omit) }}"
+    public: "{{ item.public | default(True) | bool }}"
+  with_items: "{{ os_container_clusters_templates }}"
+
+- name: Hide cluster templates if specified
+  command: "{{ os_container_clusters_venv }}/bin/openstack coe cluster template update {{ item.name }} replace hidden=True"
```

Comment:
This is new. You need to do the same as the old code.

```
    {{ os_container_clusters_venv ~ '/bin/' if os_container_clusters_venv else '' }}openstack
    {% for auth_name, auth_value in os_container_clusters_auth.items() %}
    --os-{{ auth_name | replace('_', '-') }}='{{ auth_value }}'
    {% endfor %}
    {% if os_container_clusters_cacert is defined %}
    --os-cacert {{ os_container_clusters_cacert }}
    {% endif %}
    --os-interface=public
```

Reply:
@markgoddard  I don't really understand this, why is this different to the os-images style? 
https://github.com/stackhpc/ansible-role-os-images/blob/4696b3112d216c5cb2d871c6a98e91845a274147/tasks/promote.yml#L41

Reply:
I think that os-images code relies on some bad assumptions

1. having the OS_* environment variables set
2. using a local connection (rather than SSH) where the environment variables of the control host are available to the module